jgrapht-core/src/main/java/org/jgrapht/alg/matching/DefaultMatching.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * (C) Copyright 2016-2016, by Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.matching;
+
+import java.io.Serializable;
+import java.util.Set;
+
+import org.jgrapht.alg.interfaces.MatchingAlgorithm.Matching;
+
+/**
+ * A default implementation of the matching interface.
+ * 
+ * @param <E> the graph edge type
+ *
+ * @author Dimitrios Michail
+ */
+class DefaultMatching<E>
```

Comment:
There are two kinds of users: those who use existing algorithms, and those who develop customized algorithms of their own (not intended for inclusion in JGraphT).  Keeping the DTO public makes it reusable by the second kind of user, which I think is a good thing.

(For the first kind of user, they would have to explicitly downcast the algorithm result to get at the DTO, and I don't see why they would even want to do this?  So I think the public footprint benefit for the second kind of user outweighs the downside.)

Regarding "we can use only one DTO for other interfaces which are also weighted edge sets", that makes sense...in that case, if we want to use a top-level class, then the class name should be unrelated to matching.